.github/workflows/build-preview.yml

```diff
@@ -0,0 +1,47 @@
+name: Build Preview
+
+on:
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+jobs:
+  build-preview:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '18'
```

Comment:
Since iD has [a .npmrc file](https://github.com/openstreetmap/iD/blob/develop/.nvmrc) which should be the single source of truth on which node to use, it might be a good idea to use something like https://github.com/osmberlin/parkraum.osm-verkehrswende.org/blob/main/.github/workflows/astro.yml#L57 which I used based on this comment https://github.com/actions/setup-node/issues/32#issuecomment-1003854758.

Reply:
thanks for the suggestion. it's updated in b8ae0d7a61efd6105c6a864b29c4b614aea02cb7